nixos/tests/installer.nix

```diff
@@ -68,160 +67,180 @@ let
                   , testCloneConfig
                   }:
     let
+
       iface = if grubVersion == 1 then "ide" else "virtio";
       isEfi = bootLoader == "systemd-boot" || (bootLoader == "grub" && grubUseEfi);
 
-      # FIXME don't duplicate the -enable-kvm etc. flags here yet again!
-      qemuFlags =
-        (if system == "x86_64-linux" then "-m 768 " else "-m 512 ") +
-        (optionalString (system == "x86_64-linux") "-cpu kvm64 ") +
-        (optionalString (system == "aarch64-linux") "-enable-kvm -machine virt,gic-version=host -cpu host ");
+      makeMachineConfig = name: pkgs.writeText "machine-config.json" (builtins.toJSON ({
+        inherit name;
+        # FIXME don't duplicate the -enable-kvm etc. flags here yet again!
+        qemuFlags =
+          (if system == "x86_64-linux" then "-m 768 " else "-m 512 ") +
+          (optionalString (system == "x86_64-linux") "-cpu kvm64 ") +
+          (optionalString (system == "aarch64-linux") "-enable-kvm -machine virt,gic-version=host -cpu host ");
+
+        hda = "vm-state-machine/machine.qcow2";
+        hdaInterface = iface;
+      } // (optionalAttrs isEfi {
+        bios = if pkgs.stdenv.isAarch64
+          then "${pkgs.OVMF.fd}/FV/QEMU_EFI.fd"
+          else "${pkgs.OVMF.fd}/FV/OVMF.fd";
+      })));
 
-      hdFlags = ''hda => "vm-state-machine/machine.qcow2", hdaInterface => "${iface}", ''
-        + optionalString isEfi (if pkgs.stdenv.isAarch64
-            then ''bios => "${pkgs.OVMF.fd}/FV/QEMU_EFI.fd", ''
-            else ''bios => "${pkgs.OVMF.fd}/FV/OVMF.fd", '');
     in if !isEfi && !(pkgs.stdenv.isi686 || pkgs.stdenv.isx86_64) then
       throw "Non-EFI boot methods are only supported on i686 / x86_64"
     else ''
+      def create_machine_from_file(machine_config_file):
+          with open(machine_config_file, "r") as f:
+              return create_machine(json.load(f))
+
 
-      $machine->start;
+      machine.start()
 
       # Make sure that we get a login prompt etc.
-      $machine->succeed("echo hello");
-      #$machine->waitForUnit('getty@tty2');
-      #$machine->waitForUnit("rogue");
-      $machine->waitForUnit("nixos-manual");
+      machine.succeed("echo hello")
+      # machine.wait_for_unit("getty@tty2")
+      # machine.wait_for_unit("rogue")
```

Comment:
it looks like "echo hello" is a leftover from earlier experiments. is this and the commented out lines going to vanish before this is merged?

Reply:
I agree.